### PR TITLE
kubernetes: deployment pod table

### PR DESCRIPTION
**Pull request by mclarke47**
## Hey, I just made a Pull Request!

Part of https://github.com/spotify/backstage/issues/2553

Adding an initial component to show some pod details for each deployment for each cluster.

We're (hopefully) going to have a design session at some point to get a better UI in place, but let's add this in for now so that people can at least try the plugin end to end.

Probably need to add some more tests and assertions.

### When everything is going well

Screen Shot 2020-09-25 at 15 45 55

### When everything is going not so well

Screen Shot 2020-09-25 at 15 46 01

#### :heavy_check_mark: Checklist

- [x] All tests are passing `yarn test`
- [x] Screenshots attached (for UI changes)
- [x] Relevant documentation updated
- [x] Prettier run on changed files
- [x] Tests added for new functionality
- [x] Regression tests added for bug fixes

### CODE REVIEW

File: plugins/kubernetes/src/components/DeploymentTables/DeploymentTables.tsx [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+  const isOwnedBy = (
+    ownerReferences: V1OwnerReference[],
+    obj: V1Pod | V1ReplicaSet | V1Deployment,
+  ): boolean => {
+    return ownerReferences?.some(or => or.name === obj.metadata?.name);
+  };
+
+  return (
+    <Grid
+      style={{ width: '100%' }}
```
Comment (Rugvip) [inline plugins/kubernetes/src/components/DeploymentTables/DeploymentTables.tsx:184]: Shouldn't be needed should it? ^_>

We don't want this to be needed at least :grin:

File: plugins/kubernetes/src/components/KubernetesContent/KubernetesContent.tsx [hunk 2]
```diff
@@ -45,27 +85,22 @@ export const KubernetesContent: FC<{ entity: Entity }> = ({ entity }) => {
...
-                          .map((v: any) => v.metadata.name)
-                          .join(' ')}
-                      </div>
-                    ))}
-                  </InfoCard>
-                </Grid>
-              ))}
-            </div>
-          )}
+          {kubernetesObjects !== undefined &&
```
Comment (Rugvip) [inline plugins/kubernetes/src/components/KubernetesContent/KubernetesContent.tsx:90]: Might be nicer to `kubernetesObjects?.items.map(...)` here

File: plugins/kubernetes/src/components/KubernetesContent/KubernetesContent.tsx [hunk 3]
```diff
@@ -45,27 +85,22 @@ export const KubernetesContent: FC<{ entity: Entity }> = ({ entity }) => {
...
+            kubernetesObjects.items.map((item, i) => (
+              <Grid item key={i}>
+                <InfoCard
+                  key={item.cluster.name}
```
Comment (Rugvip) [inline plugins/kubernetes/src/components/KubernetesContent/KubernetesContent.tsx:94]: Shouldn't need a key here right?

**Review by Rugvip — APPROVED** [review]
🎉

File: plugins/kubernetes/src/components/KubernetesContent/KubernetesContent.tsx (hunk 3 above)
Comment (mclarke47) [reply plugins/kubernetes/src/components/KubernetesContent/KubernetesContent.tsx:94]: Yup, forgot to remove

File: plugins/kubernetes/src/components/KubernetesContent/KubernetesContent.tsx (hunk 2 above)
Comment (mclarke47) [reply plugins/kubernetes/src/components/KubernetesContent/KubernetesContent.tsx:90]: sounds good

File: plugins/kubernetes/src/components/DeploymentTables/DeploymentTables.tsx (hunk 1 above)
Comment (mclarke47) [reply plugins/kubernetes/src/components/DeploymentTables/DeploymentTables.tsx:184]: Must have forgot to remove 😉

**Pull request merged**